dali/operators/reader/loader/video/nvdecode/ColorSpace.cu

```diff
@@ -0,0 +1,83 @@
+/*
+* Copyright 2017-2021 NVIDIA Corporation.  All rights reserved.
+*
+* Please refer to the NVIDIA end user license agreement (EULA) associated
+* with this source code for terms and conditions that govern your use of
+* this software. Any use, reproduction, disclosure, or distribution of
+* this software and related documentation outside the terms of the EULA
+* is strictly prohibited.
+*
+*/
+
+#include "ColorSpace.h"
+
+typedef struct {
+    uint8_t r, g, b;
+} Rgb;
+
+
+__constant__ float matYuv2Rgb[3][3] = {
+    1.164383f,  0.0f,       1.596027f,
+    1.164383f, -0.391762f, -0.812968f,
+    1.164383f,  2.017232f,  0.0f
+};
+
+__device__ static uint8_t Clamp(float x, float lower, float upper) {
+    return x < lower ? lower : (x > upper ? upper : x);
+}
+
+__device__ inline Rgb YuvToRgbForPixel(uint8_t y, uint8_t u, uint8_t v) {
+    const int 
+        low = 1 << (sizeof(uint8_t) * 8 - 4),
+        mid = 1 << (sizeof(uint8_t) * 8 - 1);
+    float fy = (int)y - low, fu = (int)u - mid, fv = (int)v - mid;
+    const float maxf = (1 << sizeof(uint8_t) * 8) - 1.0f;
+
+    return Rgb { 
+        Clamp(matYuv2Rgb[0][0] * fy + matYuv2Rgb[0][1] * fu + matYuv2Rgb[0][2] * fv, 0.0f, maxf),
+        Clamp(matYuv2Rgb[1][0] * fy + matYuv2Rgb[1][1] * fu + matYuv2Rgb[1][2] * fv, 0.0f, maxf),
+        Clamp(matYuv2Rgb[2][0] * fy + matYuv2Rgb[2][1] * fu + matYuv2Rgb[2][2] * fv, 0.0f, maxf)};
+}
+
+__global__ static void YuvToRgbKernel(uint8_t *pYuv, int nYuvPitch, uint8_t *pRgb, int nRgbPitch, int nWidth, int nHeight) {

```

Comment:
I wonder if this can be moved to color_space_conversion_impl.h.

Reply:
I made this file for readable for now.

In the future, we need to extend this simple code to support more pixel formats. Then I think will be the right time to extract impl. Is that ok?

Reply:
Sure